zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -122,8 +122,13 @@ fn init_wallet_db_internal<P: consensus::Parameters + 'static>(
     seed: Option<SecretVec<u8>>,
     target_migration: Option<Uuid>,
 ) -> Result<(), MigratorError<WalletMigrationError>> {
+    // Turn off foreign keys, and ensure that table replacement/modification
+    // does not break views
     wdb.conn
-        .execute("PRAGMA foreign_keys = OFF", [])
+        .execute_batch(
+            "PRAGMA foreign_keys = OFF;
+                        PRAGMA legacy_alter_table = TRUE;",
```

Comment:
```suggestion
            "PRAGMA foreign_keys = OFF;
            PRAGMA legacy_alter_table = TRUE;",
```

I also get suspicious when flags named `legacy_*` are being enabled; I'd prefer some more documentation explaining why this is necessary, and why it won't break with some future version of SQLite.